### PR TITLE
Enhance autoindent

### LINKED ISSUE #874: Enhance Autoindent
Auto indent the next line when:
- The line is `var`, `let` or `const`
- The line finishes with `or`, `and` or `object`
- The instruction isn't complete (eg: Parenthesis aren't closed)

And this: (`<>` represents the indentation)
```rust
fn main() {}
```
(Cursor is between `{` and `}`)
`<Return>`
```rust
fn main() {
<>
}
```

or

```nim
echo someFunc()
```
(Cursor is between `(` and `)`)
`<Return>`
```nim
echo someFunc(
<>
)
```

See https://github.com/zah/nim.vim

### PR DESCRIPTION
**Pull request by fox0430**
Enhance auto-indent for some languages (Nim, C, C++, C#, Java, JS, Yaml).

Close #874

### CODE REVIEW

File: src/moepkg/editor.nim [hunk 1]
```diff
@@ -279,19 +251,259 @@ proc keyEnter*(bufStatus: var BufferStatus,
...
+    line = bufStatus.buffer[currentLine]
+
+  if line.len > 0:
+    # Auto indent if the current line are "var", "let", "const".
+    # And, if finish the current line with ':', "object"
+    if (currentColumn == line.len) and
+       (line.splitWhitespace == @[ru "var"] or
+       line.splitWhitespace == @[ru "let"] or
+       line.splitWhitespace == @[ru "const"] or
+       (line.len > 6 and line[line.len - 6 .. ^1] == ru "object") or
```
Comment (uninhm) [inline src/moepkg/editor.nim:322]: Why not:
`line.splitWhitespace[^1] == (ru "object") or`

Comment (fox0430) [reply src/moepkg/editor.nim:322]: You are right. This should be ```line.splitWhitespace[^1] == (ru "object") or```

Comment (uninhm) [reply src/moepkg/editor.nim:322]: Solved #1422

Comment (fox0430) [reply src/moepkg/editor.nim:322]: Thanks!

**Pull request merged**